File: OsmAnd/src/net/osmand/plus/myplaces/tracks/filters/SingleFieldTrackFilterParams.java
package net.osmand.plus.myplaces.tracks.filters;

import android.graphics.drawable.Drawable;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import net.osmand.plus.OsmandApplication;

public class SingleFieldTrackFilterParams {
	boolean hasSelectAllVariant() {
		return false;
	}

	String getItemText(OsmandApplication app, String itemName) {
		return itemName;
	}

	@Nullable
	Drawable getItemIcon(OsmandApplication app, String itemName) {
		return null;
	}

	@Nullable
	Drawable getAllItemsIcon(OsmandApplication app, boolean isChecked, boolean nightMode) {
		return null;
	}

	@NonNull
	String trackParamToString(@NonNull Object trackParam) {
		return trackParam.toString();
	}

	public boolean includeEmptyValues() {
		return false;
	}

	public boolean sortByName() {
		return false;
	}

	public boolean sortDescending() {
		return true;
	}
}

File: OsmAnd-shared/src/commonMain/kotlin/net/osmand/shared/filters/SingleFieldTrackFilterParams.kt
package net.osmand.plus.myplaces.tracks.filters


open class SingleFieldTrackFilterParams {
	open fun hasSelectAllVariant(): Boolean {
		return false
	}

	open fun getItemText(itemName: String): String {
		return itemName
	}

//	@Nullable
//	fun getItemIcon(
//		app: OsmandApplication?,
//		itemName: String?): android.graphics.drawable.Drawable? {
//		return null
//	}
//
//	@Nullable
//	fun getAllItemsIcon(
//		app: OsmandApplication?,
//		isChecked: Boolean,
//		nightMode: Boolean): android.graphics.drawable.Drawable? {
//		return null
//	}

	open fun trackParamToString(trackParam: Any): String {
		return trackParam.toString()
	}

	open fun includeEmptyValues(): Boolean {
		return false
	}

	open fun sortByName(): Boolean {
		return false
	}

	open fun sortDescending(): Boolean {
		return true
	}
}